drivers/gpio/gpio_mcux.c

```diff
@@ -32,19 +42,23 @@ struct gpio_mcux_data {
 	sys_slist_t callbacks;
 };
 
-static int gpio_mcux_configure(const struct device *dev,
-			       gpio_pin_t pin, gpio_flags_t flags)
+static int gpio_mcux_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
 {
 	const struct gpio_mcux_config *config = dev->config;
 	GPIO_Type *gpio_base = config->gpio_base;
+
+#if defined(CONFIG_PINCTRL_NXP_IOCON)
```

Comment:
Perhaps rather than doing all this `#ifdef`'ery, could we split the pin control settings into two functions?
- `gpio_mcux_port_configure`
- `gpio_mcux_iopctl_configure`

Then we could simply have one `ifdef` in this function, and call the correct helper

Reply:
Split to two functions. Updated.